docs/decisions/2023-11-20_auto_retry_tasks.md

```diff
@@ -0,0 +1,44 @@
+# 2023-11-20 Auto Retry Tasks
+
+* status: accepted
+* date: 2023-11-20
+* authors: Malik Hadjri
+
+## Context and Problem Statement
+
+It was requested to add a feature to Evergreen to allow users to automatically retry tasks that fail due to a transient error. This
+would allow users to avoid having to manually retry these tasks, which can happen at a a semi-common
+rate depending on the task environment.
+
+## Considered Options
+
+* Create a new REST endpoint (or repurpose an existing one) that allows for tasks to mark themselves to restart upon failure,
+with the expectation that interested teams would request service users with the needed permissions to hit this endpoint, at which
+point a task could arbitrarily decide when to mark itself for restart via shell scripts 
+
+* Use the agent's existing local API, by adding a retryable parameter to its custom end task response route that users already
+leverage to let tasks define their own task end status. This new retryable parameter would be forwarded over to the Evergreen
+app server when setting the custom end task response and would trigger an automatic restart of the task in the end task logic
+on the app server side
+
+* Introduce a new agent command named something like `auto.retry` which will make the agent hit an app server endpoint
+that will mark the task for restart upon failure
+
+* Add a new boolean parameter on the generic command struct, such that if that command fails and the flag is set, we send a
+request to the app server to mark the task for restart upon completion
+
+## Decision Outcome
+
+Option 1 was rejected because it would require interested teams to request service users with the needed permissions to hit the
```

Comment:
I'm actually a little concerned about this outcome because it seems like this is a reason we're pushing for Self Service Users. Are there other reasons this would cause issues?

Reply:
Ideally Self Service Users will be getting some permissions that we'd carefully check don't have risk to cause potential harm before allowing them to be autoprovisioned (just read operations. + annotation write operations).

Whereas a service user that has the power to do something as high-level as restarting a task could be more likely to be misused by using it for other routes.